mongodb-atlas/cmd/main.go

```diff
@@ -0,0 +1,99 @@
+package main
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+
+	"github.com/icholy/digest"
+	//mongodbatlasplugin "github.com/opencost/opencost-plugins/mongodb-atlas/mongodbatlasplugin"
+)
+
+// TODO move these types to mongodbatlasplugin package
+type CreateCostExplorerQueryPayload struct {
+	Clusters              []string `json:"clusters"`
+	EndDate               string   `json:"endDate"`
+	GroupBy               string   `json:"groupBy"`
+	IncludePartialMatches bool     `json:"includePartialMatches"`
+	Organizations         []string `json:"organizations"`
+	Projects              []string `json:"projects"`
+	Services              []string `json:"services"`
+	StartDate             string   `json:"startDate"`
+}
+
+type CreateCostExplorerQueryResponse struct {
+	token string
+}
+
+type MongoDBAtlasConfig struct {
+	PublicKey  string
+	PrivateKey string
+}
+
+func main() {
+	fmt.Println("Initialize plugin")
+	url := "https://cloud.mongodb.com/api/atlas/v2/orgs/5c06ddb279358e17bfcfca49/billing/costExplorer/usage"
+	payload := CreateCostExplorerQueryPayload{
+
+		EndDate:       "2024-07-01",
+		StartDate:     "2023-12-01",
+		Organizations: []string{"5c06ddb279358e17bfcfca49"},
+	}
+	config := MongoDBAtlasConfig{
+		PublicKey:  os.Getenv("mapuk"),
+		PrivateKey: os.Getenv("maprk"),
+	}
+	resp, error := createCostExplorerQueryToken(url, payload, config)
+	fmt.Println(string(resp))
+	fmt.Println(error)
+}
+
+func getTruth() bool {
+	return true
+}
+
+// pass list of orgs , start date, end date
+func createCostExplorerQueryToken(url string, payload CreateCostExplorerQueryPayload,
+	atlasAuthConfig MongoDBAtlasConfig) (string, error) {
+
+	payloadJson, marshallingError := json.Marshal(payload)
+	if marshallingError != nil {
+		return "error", marshallingError
+	}
+	fmt.Println(string(payloadJson))
+
+	client := &http.Client{
+		Transport: &digest.Transport{
+			Username: atlasAuthConfig.PublicKey,
+			Password: atlasAuthConfig.PrivateKey,
+		},
+	}
+	request, error := http.NewRequest("POST", url, bytes.NewBuffer(payloadJson))
+
+	if error != nil {
+		return "unable to create request", error
+	}
+
+	request.Header.Set("Accept", "application/vnd.atlas.2023-01-01+json")
+	request.Header.Set("Content-Type", "application/vnd.atlas.2023-01-01+json")
+
+	response, error := client.Do(request)
+	if error != nil {
+		return response.Status, error
+	}
+	defer response.Body.Close()
+
+	fmt.Println("response Status:", response.Status)
```

Comment:
I'd go ahead and move all this to a debug or trace level logger. This will be useful for ongoing maintenance